src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -74,7 +74,7 @@ public Http2OutputProducer(Http2Stream stream, Http2StreamContext context, Strea
             // The minimum output data rate is enforced at the connection level by Http2FrameWriter.
             _flusher = new TimingPipeFlusher(_pipeWriter, timeoutControl: null, _log);
 
-            _dataWriteProcessingTask = ProcessDataWrites();
+            _dataWriteProcessingTask = ProcessDataWrites().Preserve();
```

Comment:
`ProcessDataWrites` will always return an incomplete task. Should the method just return `Task`?

(doesn't need to happen in this PR. I want to figure out if we're making things more complicated for ourselves than is nessessary by using VT in some places)

Reply:
Makes sense to me.

Reply:
On the other hand, even though this is a per-connection allocation. It's a 100-per-connection allocation. Maybe it is worth keeping it as a ValueTask for the potential "async ValueTask" caching (https://github.com/dotnet/runtime/issues/13633) benefits.

But then again, 100-per-connection isn't a whole lot either.

Any thoughts @stephentoub?

Reply:
I think it should return a Task. The ValueTask is backed by a Task anyways.